db.go

```diff
@@ -528,6 +530,38 @@ func (db *DB) Close() error {
 	return merr.Err()
 }
 
+// ToggleCompactions toggles compactions and returns if compactions are on or not.
+func (db *DB) ToggleCompactions() bool {
+	if db.compacting {
+		db.cmtx.Lock()
+		db.compacting = false
+		return false
+	}
+
+	db.cmtx.Unlock()
+	db.compacting = true
+	return true
+}
+
+// Snapshot writes the current headBlock snapshots to snapshots directory.
+func (db *DB) Snapshot(dir string) error {
+	db.mtx.Lock() // To block any appenders.
+	defer db.mtx.Unlock()
+
+	db.cmtx.Lock()
+	defer db.cmtx.Unlock()
+
+	blocks := db.blocks[:]
+	for _, b := range blocks {
+		db.logger.Log("msg", "compacting block", "block", b.Dir())
```

Comment:
Maybe "snapshotting block" so we can distinguish in our logs from regular compactions.